crates/router/src/routes/app.rs

```diff
@@ -172,6 +176,14 @@ impl AppState {
                 .await
                 .expect("Failed to create event handler");
 
+            #[allow(clippy::expect_used)]
+            #[cfg(feature = "olap")]
+            let opensearch_client = conf
+                .opensearch
+                .get_opensearch_client()
+                .await
+                .expect("Failed to create opensearch client");
```

Comment:
do we not start the application if opensearch is not set up? Only if this is enabled via a config, then we can fail the application startup